lib/discordrb/data/emoji.rb

```diff
@@ -8,18 +8,17 @@ class Emoji
     # @return [String] the emoji name
     attr_reader :name
 
-    # @return [Server] the server of this emoji
+    # @return [Server,] the server of this emoji
     attr_reader :server
 
-    # @return [Array<Role>] roles this emoji is active for
+    # @return [Array<Role>, nil] roles this emoji is active for
     attr_reader :roles
 
     # @return [true, false] if the emoji is animated
     attr_reader :animated
     alias_method :animated?, :animated
 
-    def initialize(data, bot, server)
-      @bot = bot
+    def initialize(data, server = nil)
```

Comment:
This shouldn't even be public. Don't care if you keep the default, but it should be hidden.
(We have quite a few visible private ctors in the lib that need this treatment)
```suggestion
    # @!visibility private
    def initialize(data, server)
```